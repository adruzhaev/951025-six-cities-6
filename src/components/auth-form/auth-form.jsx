import React from 'react';
import PropTypes from 'prop-types';
import {login} from '../../store/auth/api-actions';
import {connect} from 'react-redux';

const AuthForm = ({onSubmit}) => {

  const handleSubmit = (evt) => {
    evt.preventDefault();

    const data = {};
    const fields = new FormData(evt.target);

    for (let [name, value] of fields.entries()) {
      data[name] = value;
    }

    onSubmit(data);
  };

  return (
    <form className="login__form form" action="#" method="post" onSubmit={handleSubmit}>
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">E-mail</label>
        <input
          className="login__input form__input"
          type="email"
          name="email"
          id="email"
          placeholder="Email"
          required
        />
      </div>
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">Password</label>
        <input
          className="login__input form__input"
          type="password"
          name="password"
          id="password"
          placeholder="Password"
          required
        />
      </div>
      <button
        className="login__submit form__submit button"
        type="submit"
      >
                Sign in
      </button>
    </form>
  );
};

AuthForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit(authData) {
    dispatch(login(authData));
  }
});

export {AuthForm};
export default connect(null, mapDispatchToProps)(AuthForm);
