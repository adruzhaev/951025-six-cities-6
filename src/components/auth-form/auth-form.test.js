import React from 'react';
import {render} from '@testing-library/react';
import {AuthForm} from './auth-form';

it(`AuthForm should render correctly`, () => {
  const {container} = render(
      <AuthForm />
  );
  expect(container).toMatchSnapshot();
});
