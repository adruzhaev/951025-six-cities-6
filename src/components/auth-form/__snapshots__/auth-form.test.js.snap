// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm should render correctly 1`] = `
<div>
  <form
    action="#"
    class="login__form form"
    method="post"
  >
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
        for="email"
      >
        E-mail
      </label>
      <input
        class="login__input form__input"
        id="email"
        name="email"
        placeholder="Email"
        required=""
        type="email"
      />
    </div>
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
        for="password"
      >
        Password
      </label>
      <input
        class="login__input form__input"
        id="password"
        name="password"
        placeholder="Password"
        required=""
        type="password"
      />
    </div>
    <button
      class="login__submit form__submit button"
      type="submit"
    >
      Sign in
    </button>
  </form>
</div>
`;
